### PR TITLE
do not require clientId in user search to support fabric.auth searche…

### PR DESCRIPTION
**Pull request by bsmithvt**
…s for users in roles that are shared across clients

---
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/175)

### CODE REVIEW

File: Fabric.Identity.API/Management/UsersController.cs [hunk 1]
```diff
@@ -102,26 +102,21 @@ public IActionResult Search(string searchText, string identityProvider)
...
+            if (!string.IsNullOrEmpty(clientId))
             {
-                return CreateFailureResponse($"The specified client with id: {clientId} was not found",
-                    HttpStatusCode.NotFound);
+                var client = _clientManagementStore.FindClientByIdAsync(clientId).Result;
+                if (string.IsNullOrEmpty(client?.ClientId))
+                {
+                    return CreateFailureResponse($"The specified client with id: {clientId} was not found",
+                        HttpStatusCode.NotFound);
+                }
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Management/UsersController.cs]: What happens in line 124 when clientId is null or empty when we call ToUserViewModel?

Comment (bsmithvt) [reply Fabric.Identity.API/Management/UsersController.cs]: A null `LastLoginDate` is returned. Below is the implementation.

```
 public static UserApiModel ToUserViewModel(this User user, string clientId)
        {
            var lastLoginForClient = user.LastLoginDatesByClient
                .SingleOrDefault(l => l.ClientId.Equals(clientId, StringComparison.OrdinalIgnoreCase));

            DateTime? dateToSet = lastLoginForClient?.LoginDate;

            return new UserApiModel
            {
                SubjectId = user.SubjectId,
                FirstName = user.FirstName,
                LastName = user.LastName,
                MiddleName = user.MiddleName,
                LastLoginDate = dateToSet
            };
        }
```

Comment (bsmithvt) [reply Fabric.Identity.API/Management/UsersController.cs]: I'll add a check to the beginning of that method so we bypass all of the "last login date" logic if `clientId` is null or empty.

Comment (michaelvidal24) [reply Fabric.Identity.API/Management/UsersController.cs]: Thanks.

**Comment by kpaul** [comment]
just a general comment, we will want to update the docs as well for this endpoint now that client id is not required

**Pull request merged**